From c650c9473db32fc2197a32017b0ef3406cf75280 Mon Sep 17 00:00:00 2001
From: Eike Ziller <eike.ziller@qt.io>
Date: Fri, 18 Dec 2020 15:07:10 +0100
Subject: [PATCH] cmake build: Don't install highlighting files when using
 system lib

When using the system provided KF5SyntaxHighlighting library,
the highlighting files are taken from the system too.

Fixes: QTCREATORBUG-25143
Change-Id: Iac0b9734bc96e1f72d1494622081da7e279e5554
Reviewed-by: Cristian Adam <cristian.adam@qt.io>
---
 .../syntax-highlighting/CMakeLists.txt        | 20 ++++++++++---------
 1 file changed, 11 insertions(+), 9 deletions(-)

diff --git a/src/libs/3rdparty/syntax-highlighting/CMakeLists.txt b/src/libs/3rdparty/syntax-highlighting/CMakeLists.txt
index 062ae058c3..8aadf8525f 100644
--- a/src/libs/3rdparty/syntax-highlighting/CMakeLists.txt
+++ b/src/libs/3rdparty/syntax-highlighting/CMakeLists.txt
@@ -44,13 +44,15 @@ add_qtc_library(KSyntaxHighlighting SHARED
 )
 qtc_add_public_header(autogenerated/src/lib/State)
 
-install(
-  DIRECTORY data/syntax
-  DESTINATION "${IDE_DATA_PATH}/generic-highlighter/"
-)
+if(TARGET KSyntaxHighlighting)
+  install(
+    DIRECTORY data/syntax
+    DESTINATION "${IDE_DATA_PATH}/generic-highlighter/"
+  )
 
-# copy resource directories during build
-qtc_copy_to_builddir(copy_generic_highligher_to_builddir
-  DIRECTORIES data/syntax
-  DESTINATION "${IDE_DATA_PATH}/generic-highlighter/syntax"
-)
+  # copy resource directories during build
+  qtc_copy_to_builddir(copy_generic_highligher_to_builddir
+    DIRECTORIES data/syntax
+    DESTINATION "${IDE_DATA_PATH}/generic-highlighter/syntax"
+  )
+endif()
-- 
2.29.2

